### PR TITLE
alts: create handshaker RPC lazily

### PR DESCRIPTION
**Pull request by jiangtaoli2016**
This PR fixes bug described in https://github.com/grpc/grpc-java/issues/7622

### CODE REVIEW

File: alts/src/main/java/io/grpc/alts/internal/AltsHandshakerStub.java [hunk 1]
```diff
@@ -29,28 +29,29 @@
...
   private final AtomicReference<String> exceptionMessage = new AtomicReference<>();
 
   private static final long HANDSHAKE_RPC_DEADLINE_SECS = 20;
 
   AltsHandshakerStub(HandshakerServiceStub serviceStub) {
-    this.writer =
-        serviceStub
-            .withDeadlineAfter(HANDSHAKE_RPC_DEADLINE_SECS, SECONDS)
-            .doHandshake(this.reader);
+    writer = null;
```
Comment (ejona86) [inline alts/src/main/java/io/grpc/alts/internal/AltsHandshakerStub.java:41]: Delete this line. It does nothing. Initialization is only required for `final` fields. Other fields default to 0/null. Ditto below.

File: alts/src/main/java/io/grpc/alts/internal/AltsHandshakerStub.java [hunk 2]
```diff
@@ -72,6 +74,14 @@ public HandshakerResp send(HandshakerReq req) throws InterruptedException, IOExc
     return result.get();
   }
 
+  /** Create a new writer if the writer is null. */
+  private synchronized void createWriterIfNull() {
```
Comment (ejona86) [inline alts/src/main/java/io/grpc/alts/internal/AltsHandshakerStub.java:78]: Why is this synchronized? I don't think this class can be used from multiple callers/threads.

Comment (jiangtaoli2016) [reply alts/src/main/java/io/grpc/alts/internal/AltsHandshakerStub.java]: You are right. I was just being cautious :)  synchronized removed.

**Comment by jiangtaoli2016** [comment]
@ejona86 thanks for quick review. Code revised. PTAL.

**Review by ejona86 — APPROVED** [review]
Thanks!

**Pull request merged**